flow/layers/layer_tree.cc

```diff
@@ -58,9 +64,42 @@ bool LayerTree::Preroll(CompositorContext::ScopedFrame& frame,
   };
 
   root_layer_->Preroll(&context, frame.root_surface_transformation());
+
+  TryToRasterCache(&context);
+
   return context.surface_needs_readback;
 }
 
+void LayerTree::TryToRasterCache(PrerollContext* context,
+                                 bool ignore_raster_cache) {
+  auto mutator_stack = MutatorsStack();
+  for (unsigned i = 0; i < context->raster_cached_entries.size(); i++) {
+    auto& entry = context->raster_cached_entries[i];
+    if (entry->need_caching) {
+      PrerollContext preroll_context = {
+          // clang-format off
+          .raster_cache                  = context->raster_cache,
+          .gr_context                    = context->gr_context,
+          .view_embedder                 = context->view_embedder,
+          .mutators_stack                = mutator_stack,
+          .dst_color_space               = context->dst_color_space,
+          .cull_rect                     = entry->cull_rect,
+          .surface_needs_readback        = false,
+          .raster_time                   = context->raster_time,
+          .ui_time                       = context->ui_time,
+          .texture_registry              = context->texture_registry,
+          .checkerboard_offscreen_layers = context->checkerboard_offscreen_layers,
+          .frame_device_pixel_ratio      = context->frame_device_pixel_ratio,
+          .has_platform_view             = entry->has_platform_view,
+          .has_texture_layer             = entry->has_texture_layer,
+          // clang-format on
+      };
+      entry->TryToPrepareRasterCache(&preroll_context);
```

Comment:
This can fail for a number of reasons and yet you always skip the children. This method should return a true/false success status and the children should only be skipped if it fails.

Reply:
Even if it succeeds, the children may want to touch their entries, but I think we need some basic logic change in the raster cache to do this. They probably shouldn't keep their cached images for too long after they are superseded by a parent's cache, but they may want to mark their "Entry" in the raster cache so that their eligibility for caching remains valid - that way if the parent stops caching they can step in and cache themselves without having to count up from 0 accesses.